auth_backends/views.py

```diff
@@ -41,21 +42,43 @@ def dispatch(self, request, *args, **kwargs):
 
         return super(LogoutRedirectBaseView, self).dispatch(request, *args, **kwargs)
 
+    def get_redirect_url(self, *args, **kwargs):
+        strategy = load_strategy(self.request)
+        backend = load_backend(strategy, self.auth_backend_name, None)
+        return backend.logout_url
 
-class EdxOpenIdConnectLoginView(RedirectView):
+
+class LoginRedirectBaseView(RedirectView):
+    """ Base view for backend logins. """
+    auth_backend_name = None
+    permanent = False
+    query_string = True
+
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse('social:begin', args=[self.auth_backend_name])
+
+
+class EdxOpenIdConnectLoginView(LoginRedirectBaseView):
     """ Login view for projects utilizing edX OpenID Connect for single sign-on.
```

Comment:
Most services use the `auth_urlpatterns`, so adding the comment won't be of much value here.